dashboard/client/components/+network-services/service-details.tsx

```diff
@@ -7,20 +7,34 @@ import { DrawerItem, DrawerTitle } from "../drawer";
 import { Badge } from "../badge";
 import { KubeEventDetails } from "../+events/kube-event-details";
 import { KubeObjectDetailsProps } from "../kube-object";
-import { Service, serviceApi } from "../../api/endpoints";
+import { Service, ServicePort, serviceApi } from "../../api/endpoints";
 import { _i18n } from "../../i18n";
 import { apiManager } from "../../api/api-manager";
 import { KubeObjectMeta } from "../kube-object/kube-object-meta";
+import { apiBase } from "../../api"
+import { Link } from "react-router-dom";
 
 interface Props extends KubeObjectDetailsProps<Service> {
 }
 
 @observer
 export class ServiceDetails extends React.Component<Props> {
+  async portForward(event: React.MouseEvent, port: ServicePort) {
+    event.preventDefault();
+
+    const { object: service } = this.props;
+    apiBase.post(`/services/${service.getNs()}/${service.getName()}/port-forward/${port.port}`, {});
+  }
+
   render() {
     const { object: service } = this.props;
     if (!service) return;
     const { spec } = service;
+    const portLinks = service.getPorts().map((port) => {
+      return([
+        <li><Link to="" title={_i18n._(t`Open in a browser`)} onClick={(e) => this.portForward(e, port) }>{port.toString()}</Link></li>
```

Comment:
Maybe it's better to split long line for better readability.
Also, it's necessary to add `key` prop to `<li>`. I guess it can be just like li contents - `post.toString()`

Reply:
I will make necessary changes.